File: Fundamentals/ListProject/app/src/main/java/com/example/listactivity/MainActivity.java
package com.example.listactivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    ArrayList<String> arrName = new ArrayList<>(); // list of names

    Button autoCompleteActivityBtn, spinnerActivityBtn, listActivityBtn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        autoCompleteActivityBtn = findViewById(R.id.autoCompleteActivityBtn);
        spinnerActivityBtn = findViewById(R.id.spinnerActivityBtn);
        listActivityBtn = findViewById(R.id.listActivityBtn);

        autoCompleteActivityBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, AutoCompleteActivity.class);
                intent.putExtra("nameList", arrName);
                startActivity(intent);
            }
        });

        spinnerActivityBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, SpinnerActivity.class);
                intent.putExtra("nameList", arrName);
                startActivity(intent);
            }
        });

        listActivityBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this, ListActivity.class);
                intent.putExtra("nameList", arrName);
                startActivity(intent);
            }
        });

        arrName.add("Anupam");
        arrName.add("Anurag");
        arrName.add("Rahul");
        arrName.add("Ram");
        arrName.add("Sidd");
        arrName.add("Virat");
        arrName.add("Rohit");
        arrName.add("Siraj");
        arrName.add("Bumrah");
        arrName.add("DeCock");
        arrName.add("Rabada");
        arrName.add("Ponting");
        arrName.add("Steve Smith");
        arrName.add("Maxwell");
        arrName.add("Gayle");
        arrName.add("Tendulkar");
        arrName.add("Dhoni");
        arrName.add("Gautam");
        arrName.add("Leander");
        arrName.add("Ussain Bolt");
    }
}

File: MoviesApplication/app/src/main/java/com/example/moviesapp/presentation/MainActivity.kt
package com.example.moviesapp.presentation

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.example.moviesapp.R
import com.example.moviesapp.databinding.ActivityMainBinding
import com.example.moviesapp.presentation.artist.ArtistsActivity
import com.example.moviesapp.presentation.movie.MoviesActivity
import com.example.moviesapp.presentation.tvShow.TvShowsActivity

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        binding.artistBtn.setOnClickListener {
            val intent = Intent(this, ArtistsActivity::class.java)
            startActivity(intent)
        }

        binding.tvShowBtn.setOnClickListener {
            val intent = Intent(this, TvShowsActivity::class.java)
            startActivity(intent)
        }

        binding.moviesBtn.setOnClickListener {
            val intent = Intent(this, MoviesActivity::class.java)
            startActivity(intent)
        }
    }
}